builtin/push.c

```diff
@@ -338,7 +366,9 @@ static int push_with_options(struct transport *transport, int flags)
 	err = transport_push(transport, refspec_nr, refspec, flags,
 			     &reject_reasons);
 	if (err != 0)
+		fprintf(stderr, "%s", push_get_color(PUSH_COLOR_ERROR));
 		error(_("failed to push some refs to '%s'"), transport->url);
+		fprintf(stderr, "%s", push_get_color(PUSH_COLOR_RESET));
```

Comment:
I have a hunch that core Git's reviewers may suggest to add those colors as additional parameters to the `error()` call. But the proof lies in the pudding, so to say: I really would like to send this as an RFC to the Git mailing list.